src/connector/src/sink/encoder/text.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use risingwave_common::catalog::Schema;
+use risingwave_common::types::ToText;
+
+use super::RowEncoder;
+
+pub struct TextEncoder {
+    pub schema: Schema,
+    // the column must contain only one element
+    pub col_index: usize,
+}
+
+impl TextEncoder {
+    pub fn new(schema: Schema, col_index: usize) -> Self {
+        Self { schema, col_index }
+    }
+}
+
+impl RowEncoder for TextEncoder {
+    type Output = String;
+
+    fn schema(&self) -> &risingwave_common::catalog::Schema {
+        &self.schema
+    }
+
+    fn col_indices(&self) -> Option<&[usize]> {
+        Some(std::slice::from_ref(&self.col_index))
+    }
+
+    fn encode_cols(
+        &self,
+        row: impl risingwave_common::row::Row,
+        col_indices: impl Iterator<Item = usize>,
+    ) -> crate::sink::Result<Self::Output> {
+        // It is guaranteed by the caller that col_indices contains only one element
+        let mut result = String::new();
+        for col_index in col_indices {
+            let datum = row.datum_at(col_index);
+            result = datum.to_text_with_type(&self.schema.fields[col_index].data_type);
```

Comment:
Second point of https://github.com/risingwavelabs/risingwave/pull/16377#discussion_r1591827285 unresolved:
* Let's make an exception for `bool` to return `true`/`false` rather than `t`/`f` used by PostgreSQL `ToText`. (Similarly in [rw-expr](https://github.com/risingwavelabs/risingwave/blob/b4ecd456bf22afd08e3e6e2ff5c4a9b45b55d2d2/src/expr/impl/src/scalar/cast.rs#L173-L185) and [pg-proc](https://github.com/postgres/postgres/blob/REL_16_2/src/backend/utils/adt/bool.c#L157-L219))

Reply:
Oh I thought it is `to_text_with_type `'s default behavior. Let me fix.